### PR TITLE
Added API handling to  messenger

### PR DESCRIPTION
**Pull request by drfifonz**

### CODE REVIEW

File: src/bot_functions.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+def process_message(text):
+    """Processing user input, returning proper response"""
+    formatted_message = text.lower().strip()
+    if formatted_message == "test":
+        response = "Test successful"
+    return response
```
Comment (Vosloo) [inline src/bot_functions.py]: Space between return and previous line

File: src/bot_functions.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+def process_message(text):
```
Comment (Vosloo) [inline src/bot_functions.py]: Preferably add return types to methods: -> type

File: src/messenger_bot.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+from flask import Flask, request
+import requests
+from pymessenger import Bot
+
+import tokens
+from bot_functions import process_message
```
Comment (Vosloo) [inline src/messenger_bot.py]: Sort imports (use tool from VScode if you use it)

File: src/messenger_bot.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# FB_API_URL = "https://graph.facebook.com/v2.6/me/messages"
+PAGE_ACCESS_TOKEN = tokens.messenger_bot_page_access_token
+VERIFY_TOKEN = tokens.user_verify_token
+
+bot = Bot(PAGE_ACCESS_TOKEN)
+
+
+@app.route("/", methods=["POST", "GET"])
+def webhook():
+
```
Comment (Vosloo) [inline src/messenger_bot.py]: Remove the newline

File: src/messenger_bot.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    if request.method == "GET":
+        """Veryfy user tokens equality"""
+        if request.args.get("hub.verify_token") == VERIFY_TOKEN:
+            return request.args.get("hub.challenge")
```
Comment (Vosloo) [inline src/messenger_bot.py:30]: Do not hardcode values, move them to constants - applies to all hardcoded values - below as well

File: src/messenger_bot.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        else:
+            return "Not connected to fb"
+            # is returned even being sucesfully conected
```
Comment (Vosloo) [inline src/messenger_bot.py]: Delete

File: src/messenger_bot.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    elif request.method == "POST":
+        """Respond to user input"""
+        payload = request.json
+        event = payload["entry"][0]["messaging"]
```
Comment (Vosloo) [inline src/messenger_bot.py]: What is 0-th element of the entry? If it is unnecessary to split this line into multiple variables then explain what is happening with a comment.

**Review by Vosloo — CHANGES_REQUESTED** [review]

File: src/messenger_bot.py (hunk 6 above)
Comment (drfifonz) [reply src/messenger_bot.py]: Flask needs to get some response

**Pull request closed without merging**